src/client/Client.cc

```diff
@@ -318,6 +318,11 @@ Client::~Client()
   // client_lock) is held.
   std::lock_guard l{client_lock};
   tear_down_cache();
+
+  // Make sure that the Finisher thread is stopped before it being
+  // destructed, because it maybe still holding the mutex lock
+  objecter_finisher.wait_for_empty();
+  objecter_finisher.stop();
```

Comment:
these are inside lock, does not match patch title

Reply:
Hi Yan,
Do you mean the client_lock above ?
This patch is trying to unlock the mutex lock in Finisher's thread, maybe I am not correctly titled it.
Thanks

Reply:
Updated the title now.

Reply:
I think it's better to do this outside of client_lock

Reply:
Okay, it makes sense.
Fixed. Thanks.